### PR TITLE
Add introduction page to Contributing Guide

### PR DESCRIPTION
**Pull request by BD103**
As part of the Contributing Guide working group, this adds the introduction page.

This was initially written on [HackMD](https://hackmd.io/@bevy/rktu0EEuC).

Blocked by #1581.

### CODE REVIEW

File: content/contributing/getting-started/_index.md [hunk 1]
```diff
@@ -8,3 +8,75 @@ status = 'hidden'
...
+## Getting oriented
+
+Bevy, like any general-purpose game engine, is a large project! It can be a bit overwhelming to start, so here's the bird's-eye view.
+
+The [Github organization] has 2 primary repositories:
+
+[Github organization]: https://github.com/bevyengine
+
+1. [`bevy`]: This is where the engine itself lives. The bulk of development work occurs here.
+2. [`bevy-website`]: This is where the [official website] is hosted, along with the quick start, news, and other essential information.
```
Comment (alice-i-cecile) [inline content/contributing/getting-started/_index.md:70]:
```suggestion
2. [`bevy-website`]: This is where the [official website] is hosted, along with the quick start, news, contributing book, live examples and other bits of polished content we want to show our users.
```

File: content/contributing/getting-started/_index.md [hunk 2]
```diff
@@ -8,3 +8,75 @@ status = 'hidden'
...
+- **Thoughtful public interfaces over maximal configurability**: Symbols and apis should be private by default. Every public API should be thoughtfully and consistently designed. Don't expose unnecessary internal implementation details. Don't allow users to "shoot themselves in the foot". Favor one "happy path" api over multiple apis for different use cases.
+- **Welcome new contributors**: Invest in new contributors. Help them fill knowledge and skill gaps. Don't ever gatekeep Bevy development according to notions of required skills or credentials. Help new developers find their niche.
+- **Civil discourse**: We need to collectively discuss ideas and the best ideas *should* win. But conversations need to remain respectful at all times. Remember that we're all in this together. Always follow our [Code of Conduct].
+- **Test what you need to**: Write useful tests. Don't write tests that aren't useful. We *generally* aren't strict about unit testing every line of code. We don't want you to waste your time. But at the same time:
```
Comment (alice-i-cecile) [inline content/contributing/getting-started/_index.md:54]: This should be moved up: welcoming + civil discourse belong together and make a nice capstone.

**Comment by BD103** [comment]
I rebased this on top of #1581, so this is now blocked until that is merged. The introduction and layout changes depend too much on each other.

File: content/contributing/introduction.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+[`bevy-website`]: https://github.com/bevyengine/bevy-website
+[official website]: https://bevyengine.org
+
+{% callout() %}
+These repositories are owned and managed by [The Bevy Foundation], a non-profit charity dedicated to keeping Bevy free for everyone everywhere, until the end of time itself!
+
+[The Bevy Foundation]: https://bevyengine.org/foundation/
+{% end %}
+
+The `bevy` repo itself contains many smaller subcrates. Most of them can be used by themselves and many of them can be modularly replaced. Feel free to dig around and find one that interests you!
```
Comment (alice-i-cecile) [inline content/contributing/introduction.md:78]:
```suggestion
The `bevy` repo itself contains many smaller subcrates. Several of them can be used by themselves and almost all of them can be modularly replaced. Feel free to dig around and find one that interests you!
```

File: content/contributing/introduction.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+- **Don't re-invent every wheel**: As a counter to the previous point, don't re-invent everything at all costs. If there is a crate in the Rust ecosystem that is the "de-facto" standard (ex: `wgpu` or `winit`), we should heavily consider using it. Bevy should be a positive force in the ecosystem. We should drive the improvements we need into these core ecosystem crates.
+- **Rust-first**: Engine and user-facing code should optimize and encourage Rust-only workflows. Adding additional languages increases internal complexity, fractures the Bevy ecosystem, and makes it harder for users to understand the engine. Never compromise a Rust interface in the interest of compatibility with other languages.
+- **Thoughtful public interfaces over maximal configurability**: Symbols and apis should be private by default. Every public API should be thoughtfully and consistently designed. Don't expose unnecessary internal implementation details. Don't allow users to "shoot themselves in the foot". Favor one "happy path" api over multiple apis for different use cases.
```
Comment (TrialDragon) [inline content/contributing/introduction.md:47]:
```suggestion
- **Thoughtful public interfaces over maximal configurability**: Symbols and APIs should be private by default. Every public API should be thoughtfully and consistently designed. Don't expose unnecessary internal implementation details. Don't allow users to "shoot themselves in the foot". Favor one "happy path" API over multiple APIs for different use cases.
```
Uppercase the acronym API.

**Review by alice-i-cecile — APPROVED** [review]

File: content/contributing/introduction.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+- **Test what you need to**: Write useful tests. Don't write tests that aren't useful. We *generally* aren't strict about unit testing every line of code. We don't want you to waste your time. But at the same time:
+    - Most new features should have at least one minimal [example](https://github.com/bevyengine/bevy/tree/main/examples). These also serve as simple integration tests, as they are run as part of our CI process.
+    - The more complex or "core" a feature is, the more strict we are about unit tests. Use your best judgement here. We will let you know if your pull request needs more tests. We use Rust's built in [testing framework].
+- **Welcome new contributors**: Invest in new contributors. Help them fill knowledge and skill gaps. Don't ever gatekeep Bevy development according to notions of required skills or credentials. Help new developers find their niche.
+- **Civil discourse**: We need to collectively discuss ideas and the best ideas *should* win. But conversations need to remain respectful at all times. Remember that we're all in this together. Always follow our [Code of Conduct].
```
Comment (TrialDragon) [inline content/contributing/introduction.md:52]:
```suggestion
- **Civil discourse**: We need to collectively discuss ideas, and the best ideas *should* win, but conversations need to remain respectful at all times. Remember that we're all in this together and always follow our [Code of Conduct].
```
Nit-pick, non-blocking: This just seems to read better this way to me.

**Review by TrialDragon — APPROVED** [review]
LGTM.

**Pull request merged**